salt/modules/file.py

```diff
@@ -1750,7 +1751,37 @@ def _get_line_indent(src, line, indent):
             break
         idt.append(c)
 
-    return ''.join(idt) + line.strip()
+    return ''.join(idt) + line.lstrip()
+
+
+def _get_eol(line):
+    compiled = re.compile('((?<!\r)\n|\r(?!\n)|\r\n)$')
```

Comment:
move this to a constant. __eol_re = e.compile('((?<!\r)\n|\r(?!\n)|\r\n)$')  Otherwise no point in compiling it first.